contracts/MasterChef.sol

```diff
@@ -0,0 +1,294 @@
+        uint256 accRewardPerShare; // Accumulated Rewards per share, times 1e12
+        uint256 farmableSupply; // set in init, total amount of tokens farmable
+        uint256 numFarmers;
+    }
+    
+    /// @notice farm type id. Useful for back-end systems to know how to read the contract (ABI) 
+    /// as we plan to launch multiple farm types
+    uint256 public farmType = 1;
+   
+    address public farmGenerator;
+
+    FarmInfo public farmInfo;
+    
+    /// @notice information on each user than stakes LP tokens
+    mapping (address => UserInfo) public userInfo;
+
+    // Receipts root vs bool. It is used to avoid same tx submission twice.
+    mapping (bytes32 => bool) public receipts;
+
+    uint256 public constant EVENT_INDEX_IN_RECEIPT = 3; 
+    uint256 public TRANSFER_EVENT_PARAMS_INDEX_IN_RECEIPT = 1;
+
+    bytes32 public constant TRANSFER_EVENT_SIG = 0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef;
+
+    address CRYPTION_VALIDATOR_SHARE_PROXY_CONTRACT = address(0x44d2c14F79EF400D6AF53822a054945704BF1FeA);
+
+    event Deposit(address indexed user, uint256 amount);
+    event Withdraw(address indexed user, uint256 amount);
+    event EmergencyWithdraw(address indexed user, uint256 amount);
+
+    constructor( address _farmGenerator) public {
+        farmGenerator = _farmGenerator;
+    }
+
+    /**
+     * @notice initialize the farming contract. 
+     * This is called only once upon farm creation and the FarmGenerator ensures the farm has the correct paramaters
+     */
+    function init (
+        IERC20 _rewardToken, 
+        uint256 _amount,
+        IERC20 _lpToken, 
+        uint256 _blockReward, 
+        uint256 _startBlock, 
+        uint256 _endBlock, 
+        uint256 _bonusEndBlock, 
+        uint256 _bonus
+        ) external onlyOwner {
+        require(msg.sender == address(farmGenerator), 'FORBIDDEN');
+
+        // TransferHelper.safeTransferFrom(address(_rewardToken), msg.sender, address(this), _amount);
+        farmInfo.rewardToken = _rewardToken;
+        
+        farmInfo.startBlock = _startBlock;
+        farmInfo.blockReward = _blockReward;
+        farmInfo.bonusEndBlock = _bonusEndBlock;
+        farmInfo.bonus = _bonus;
+        
+        uint256 lastRewardBlock = block.number > _startBlock ? block.number : _startBlock;
+        farmInfo.lpToken = _lpToken;
+        farmInfo.lastRewardBlock = lastRewardBlock;
+        farmInfo.accRewardPerShare = 0;
+        
+        farmInfo.endBlock = _endBlock;
+        farmInfo.farmableSupply = _amount;
+    }
+
+    /**
+     * @notice Gets the reward multiplier over the given _from_block until _to block
+     * @param _from_block the start of the period to measure rewards for
+     * @param _to the end of the period to measure rewards for
+     * @return The weighted multiplier for the given period
+     */
+    function getMultiplier(uint256 _from_block, uint256 _to) public view returns (uint256) {
```

Comment:
_from_block to _fromBlock ( camel case )

Reply:
Done